staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -312,7 +312,7 @@ func PollImmediateUntil(interval time.Duration, condition ConditionFunc, stopCh
 
 // WaitFunc creates a channel that receives an item every time a test
 // should be executed and is closed when the last test should be invoked.
-type WaitFunc func(done <-chan struct{}) <-chan struct{}
+type WaitFunc func(done <-chan struct{}) (ch <-chan struct{}, cancel func())
```

Comment:
Update documentation.

I'm honestly astonished that the tests continue to pass everywhere considering this breaks the public interface.

Reply:
That means nobody uses `WaitFunc` directly. :)

Reply:
Let's move it to private then.